services/cd-service/pkg/repository/transformer_test.go

```diff
@@ -710,8 +717,638 @@ func TestCreateApplicationVersion(t *testing.T) {
 			if err == nil {
 				t.Fatalf("expected error, got none.")
 			}
-			if err.Error() != tc.expectedErrorMsg {
-				t.Fatalf("Expected different error (expected: %s, got: %s)", tc.expectedErrorMsg, err.Error())
+			var expectedErr api.CreateReleaseResponse
+			if err := prototext.Unmarshal([]byte(tc.expectedErrorMsg), &expectedErr); err != nil {
+				t.Fatalf("failed to unmarshal the expected error object: %v", err)
+			}
+			
+			var actualErr api.CreateReleaseResponse
+			if err := prototext.Unmarshal([]byte(err.Error()), &actualErr); err != nil {
+				t.Fatalf("failed to unmarshal the actual error object: %v", err)
+			}
+
+			if !proto.Equal(&expectedErr, &actualErr) {
+				t.Fatalf("Expected different error (expected: %v, got: %v)", expectedErr, actualErr)
+			}
+		})
+	}
+}
+
+func dumpFilesystemHelper(fs billy.Filesystem, path string) []string {
+	ret := make([]string, 0)
+
+	files, err := fs.ReadDir(path)
+	if err == nil {
+		for _, file := range files {
+			ret = append(ret, dumpFilesystemHelper(fs, fs.Join(path, file.Name()))...)
+		}
+	} else {
+		ret = append(ret, path)
+	}
+
+	return ret
+}
+
+func dumpFilesystem(fs billy.Filesystem) []string {
+	paths := dumpFilesystemHelper(fs, ".")
+	sort.Slice(paths, func(i, j int) bool { return paths[i] < paths[j] })
+	return paths
+}
+
+func verifyCommitPathsExist(fs billy.Filesystem, paths []string) error {
+	for _, path := range paths {
+		_, err := fs.Stat(path)
+		if err != nil {
+			return fmt.Errorf(`error verifying commit path exists. path: %s, error: %v
+			
+			directory tree: %s`, path, err, strings.Join(dumpFilesystem(fs), "\n"))
```

Comment:
Odd formatting, I'm surprised the linter does not complain. Please remove the empty line in between.